packages/legend-query-builder/src/stores/filter/QueryBuilderFilterState.ts

```diff
@@ -965,48 +1137,86 @@ export class QueryBuilderFilterState
     return Boolean(
       Array.from(this.nodes.values())
         .filter(filterByType(QueryBuilderFilterTreeConditionNodeData))
-        .map((node) => node.condition.value)
+        .map((node) =>
+          node.condition.rightConditionValue instanceof
+          FilterValueSpecConditionValueState
+            ? node.condition.rightConditionValue.value
+            : undefined,
+        )
         .filter(isNonNullable)
         .find((value) => isValueExpressionReferencedInValue(variable, value)),
     );
   }
 
+  isInvalidFilterPropertyExpressionState(
+    node: QueryBuilderFilterTreeNodeData,
+  ): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      !node.condition.propertyExpressionState.isValid
+    );
+  }
+
+  isInvalidValueSpecFilterValue(node: QueryBuilderFilterTreeNodeData): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      node.condition.rightConditionValue instanceof
+        FilterValueSpecConditionValueState &&
+      node.condition.rightConditionValue.value instanceof InstanceValue &&
+      !isValidInstanceValue(node.condition.rightConditionValue.value)
+    );
+  }
+
+  isInvalidPropertyExpressionStateFilterValue(
+    node: QueryBuilderFilterTreeNodeData,
+  ): boolean {
+    return (
+      node instanceof QueryBuilderFilterTreeConditionNodeData &&
+      node.condition.rightConditionValue instanceof
+        FilterPropertyExpressionStateConditionValueState &&
+      !node.condition.rightConditionValue.propertyExpressionState.isValid
+    );
+  }
+
   get allValidationIssues(): string[] {
     const validationIssues: string[] = [];
     Array.from(this.nodes.values()).forEach((node) => {
       if (node instanceof QueryBuilderFilterTreeConditionNodeData) {
-        if (
-          node.condition.value instanceof InstanceValue &&
-          !isValidInstanceValue(node.condition.value)
-        ) {
+        if (this.isInvalidValueSpecFilterValue(node)) {
           validationIssues.push(
             `Filter value for ${node.condition.propertyExpressionState.title} is missing or invalid`,
           );
         }
-        if (!node.condition.propertyExpressionState.isValid) {
+        if (this.isInvalidFilterPropertyExpressionState(node)) {
           validationIssues.push(
             `Derived property parameter value for ${node.condition.propertyExpressionState.title} is missing or invalid`,
           );
         }
+        if (
+          node.condition.rightConditionValue instanceof
+            FilterPropertyExpressionStateConditionValueState &&
+          this.isInvalidPropertyExpressionStateFilterValue(node)
+        ) {
+          validationIssues.push(
```

Comment:
would you mind explaining again why if the right hand of filter node is a property expression, then it is a Derived property parameter value ?

Reply:
It's not always a derived property parameter value, but if the right side is a property expression that has a required parameter that is missing, then `propertyExpressionState.isValid` will be false. You can see the `DND derived property from explorer panel to filter panel` and `DND derived property from fetch structure panel to filter panel` screen recordings for examples of where the right hand side is a property expression that has a missing required parameter (we get the red (...) icon indicating the missing parameter).